src/client/pages/Embark/index.tsx

```diff
@@ -254,37 +277,41 @@ export const EmbarkRoot: React.FunctionComponent<EmbarkRootProps> = (props) => {
       const result = await apolloClient.client.query<AngelData, AngelVariables>(
         {
           query: ANGEL_DATA_QUERY,
-          variables: { name: props.name, locale: localeIsoCode },
+          variables: { name: props.name, locale: isoLocale },
         },
       )
 
       if (result.data && result.data.angelStory) {
         setData([props.name, JSON.parse(result.data.angelStory.content)])
       }
     })()
-  }, [props.name, localeIsoCode])
+  }, [props.name, isoLocale])
 
   React.useEffect(() => {
-    if (!props.name) {
+    if (!props.name || (isQuoteCartApiEnabled && !quoteCartId)) {
       return
     }
 
+    const defaultStore = isQuoteCartApiEnabled
+      ? ({ quoteCartId } as Record<string, string>)
+      : {}
+
     const prevStore = window.sessionStorage.getItem(
       `embark-store-${encodeURIComponent(props.name)}`,
     )
 
     if (!prevStore) {
-      setInitialStore({})
+      setInitialStore(defaultStore)
       return
     }
 
     try {
       const parsedPrevStore = JSON.parse(prevStore)
-      setInitialStore(parsedPrevStore as { [key: string]: any })
+      setInitialStore({ ...parsedPrevStore, ...defaultStore })
```

Comment:
Not sure if we wanna override the `quoteCartId` here (if you e.g. reload the page after starting Embark)